db/repair_bucket.go

```diff
@@ -0,0 +1,228 @@
+package db
+
+import (
+	"encoding/json"
+
+	"io/ioutil"
+
+	"fmt"
+
+	"github.com/couchbase/go-couchbase"
+	"github.com/couchbase/sync_gateway/base"
+)
+
+// Enum for the different repair jobs (eg, repairing rev tree cycles)
+type RepairJobType string
+
+const (
+	RepairRevTreeCycles = RepairJobType("RepairRevTreeCycles")
+)
+
+// Params suitable for external (eg, HTTP) invocations to describe a RepairBucket operation
+type RepairBucketParams struct {
+	DryRun     bool              `json:"dry_run"`
+	RepairJobs []RepairJobParams `json:"repair_jobs"`
+}
+
+// Params suitable for external (eg, HTTP) invocations to describe a specific RepairJob operation
+type RepairJobParams struct {
+	RepairJobType   RepairJobType          `json:"repair_job_type"`
+	RepairJobParams map[string]interface{} `json:"repair_job_params"`
```

Comment:
Since these are nested inside the 'repair_jobs' element, should we skip the 'repair_job_' prefix on the json names?

Reply:
Fixed